assignments/02-cli-constructive.md

```diff
@@ -55,63 +56,101 @@ $ touch rng.c
 $ ./build.sh 
 cc -c rng.c
 cc -o foo foo.o rng.o
+$ ./foo 
+hello, world! your randomly chosen number is 4
 $
 ```
 
-Notice two things:
+Notice three things:
 
 * We can run `build.sh` and if nothing needs to get rebuilt, nothing will be
   rebuilt
-* We can update individual C files without recompiling other C files
-
-...
+* We can update individual C files without recompiling other C files (we do
+  need to link them at the end, though)
+* By "changed", we mean that the file's modification time is newer than its
+  target; the file contents may not have changed
+
+This is a pretty standard set of features for build systems. Many people have
+put many years into making build systems great. Because we are dealing with
+small code and don't have a lot of strange requirements, you are only going to
+peek a little bit into that yawning abyss. We hope that after writing your
```

Comment:
I'd cut "We hope …" and everything after. I think referencing Make will be lost on them at this point in the course.

Reply:
:(